src/main/java/com/adobe/cq/testing/selenium/pageobject/granite/BasePage.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright 2021 Adobe Systems Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.adobe.cq.testing.selenium.pageobject.granite;
+
+import com.adobe.cq.testing.selenium.pagewidgets.coral.CoralToast;
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.Selenide;
+import com.codeborne.selenide.SelenideElement;
+import com.codeborne.selenide.WebDriverRunner;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+
+import javax.annotation.Nonnull;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+import static com.adobe.cq.testing.selenium.pagewidgets.Helpers.waitNetworkIdled;
+import static com.adobe.cq.testing.selenium.pagewidgets.Helpers.waitDocumentLoadCompleted;
+import static com.adobe.cq.testing.selenium.pagewidgets.coral.CoralReady.waitCoralReady;
+import static com.codeborne.selenide.Selenide.$;
+
+/**
+ * Base page class for inheritance.
+ */
+public class BasePage {
```

Comment:
Shouldn't this be `abstract` if it's meant just for inheritance?

Reply:
updated